newapi/ooniapi/measurements.py

```diff
@@ -1473,3 +1497,133 @@ def get_torsf_stats() -> Response:
 
     except Exception as e:
         return jsonify({"v": 0, "error": str(e)})
+
+
+# # measurement feedback
+
+from ooniapi.auth import role_required, get_account_id, jerror
+from ooniapi.database import insert_click
+
+
+"""
+CREATE TABLE msmt_feedback
+(
+    `measurement_uid` String,
+    `account_id` String,
+    `status` String,
+    `comment` String,
+    `update_time` DateTime64(3) MATERIALIZED now64()
+)
+ENGINE = ReplacingMergeTree
+ORDER BY (measurement_uid, account_id)
+SETTINGS index_granularity = 4
+"""
+
+valid_feedback_status = [
+    "blocked",
+    "blocked.blockpage",
+    "blocked.blockpage.http",
+    "blocked.blockpage.dns",
+    "blocked.blockpage.server_side",
+    "blocked.blockpage.server_side.captcha",
+    "blocked.dns",
+    "blocked.dns.inconsistent",
+    "blocked.dns.nxdomain",
+    "blocked.tcp",
+    "blocked.tls",
+    "ok",
+    "ok.unreachable",
+    "ok.broken",
+    "ok.parked",
+]
+
+
+@api_msm_blueprint.route("/v1/measurement_feedback")
+@metrics.timer("get_msmt_feedback")
+@role_required(["admin", "user"])
+def get_msmt_feedback() -> Response:
+    """Get measurement for the curred logged user for a given measurement
+    ---
+    produces:
+      - application/json
+    consumes:
+      - application/json
+    parameters:
+      - in: body
+        required: true
+        schema:
+          type: object
+          properties:
+            measurement_uid:
+              type: string
+              description: Measurement ID
+    responses:
+      200:
+        description: status and comment or empty JSON
+    """
+
+    def jparam(name):
+        return request.json.get(name, "").strip()
+
+    log = current_app.logger
+    account_id = get_account_id()
+    measurement_uid = jparam("measurement_uid")
+
+    query = """SELECT status, comment FROM msmt_feedback FINAL
+    WHERE account_id = :account_id AND measurement_uid = :measurement_uid
+    LIMIT 1"""
+    qp = dict(account_id=account_id, measurement_uid=measurement_uid)
+    row = query_click_one_row(sql.text(query), qp)
+    if row:
+        return cachedjson("0s", **row)
+
+    return cachedjson("0s")
+
+
+@api_msm_blueprint.route("/v1/submit_measurement_feedback", methods=["POST"])
+@metrics.timer("submit_msmt_feedback")
+@role_required(["admin", "user"])
+def submit_msmt_feedback() -> Response:
+    """Submit measurement feedback. Only for registered users.
+    ---
+    produces:
+      - application/json
+    consumes:
+      - application/json
+    parameters:
+      - in: body
+        required: true
+        schema:
+          type: object
+          properties:
+            measurement_uid:
+              type: string
+              description: Measurement ID
+            status:
+              type: string
+              description: Measurement status
+              minLength: 2
+            comment:
```

Comment:
Until we have a good use-case and UI for this, I would suggest we drop the comment field from it.

Reply:
Updated

Reply:
I still see the comment field in the current branch, both in the docstring and the code implementing it.

Reply:
Oops, fixed!